tests/test_signing.py

```diff
@@ -0,0 +1,97 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+
+import tempfile
+from collections.abc import Mapping
+from pathlib import Path
+
+import pytest
+
+from mkosi.run import run
+
+from . import Image, ImageConfig
+
+pytestmark = pytest.mark.integration
+
+
+@pytest.mark.skipif(
+    not Path("/usr/bin/sqop").exists(),
+    reason="requires sqop for signing but not all distros package it (e.g. not in opensuse)",
+)
```

Comment:
This doesn't actually work, sometimes tests run with a tools tree which means we pick up tools from the tools tree instead of from the host, so checking whether the binary is installed on the host like you do here doesn't work.

With https://github.com/systemd/mkosi/pull/3109 the tools tree path will be available in the `ImageConfig` object and you can use it together with `find_binary()` to check if `sqop` is available

Reply:
That's great! I tried checking for `config.distribution` and comparing that to `Distribution.opensuse` but it ended up not good :-/